server/etcdserver/server.go

```diff
@@ -1288,6 +1301,15 @@ func (s *EtcdServer) isLeader() bool {
 	return uint64(s.ID()) == s.Lead()
 }
 
+func (s *EtcdServer) isLeaderInRaftStateMachine() bool {
+	rs := s.raftStatus()
+	// leader's raftStatus.Progress is not nil
```

Comment:
We can just `return rs.Progress != nil`?

Reply:
updated.